driver/src/main/scala/com/stratio/sparkta/driver/service/RawDataStorageService.scala

```diff
@@ -16,29 +16,33 @@
 package com.stratio.sparkta.driver.service
 
 
+import java.io
+
+import com.stratio.sparkta.sdk.{DateOperations, Event}
 import org.apache.spark.sql._
 import org.apache.spark.streaming.dstream.DStream
-
-import com.stratio.sparkta.sdk.Event
+import org.joda.time.DateTime
 
 /**
- * Created by arincon on 17/04/15.
+ * @author arincon on 17/04/15.
  */
-class RawDataStorageService(sc: SQLContext, path: String) extends Serializable {
+class RawDataStorageService(sc: SQLContext, path: String, granularity: String) extends Serializable {
 
   import sc.implicits._
 
   case class RawEvent(timeStamp: String, data: String)
 
-  def composeRawFrom(event: Event) =  event.keyMap.map(e => e._2).toSeq
+  def composeRawFrom(event: Event): Seq[io.Serializable] = event.keyMap.map(e => e._2).toSeq
+
+  def timeSuffix: String = "/"+DateOperations.dateFromGranularity(DateTime.now(), granularity)
 
-  def extractRawDataFromEvent(event: Event) = {
-    event.rawData getOrElse composeRawFrom (event)
+  def extractRawDataFromEvent(event: Event): Any = {
+    event.rawData getOrElse composeRawFrom(event)
   }
 
-  def save(raw: DStream[Event]):DStream[Event] = {
+  def save(raw: DStream[Event]): DStream[Event] = {
     raw.map(event => RawEvent(System.currentTimeMillis().toString, extractRawDataFromEvent(event).toString))
-      .foreachRDD(_.toDF().save(path,"parquet", SaveMode.Append))
+      .foreachRDD(_.toDF().save(path + timeSuffix, "parquet", SaveMode.Append))
```

Comment:
I suggest to use Constants for RawData operations but up to you ... Look over HttpConstant if you want to change it.